whois-nrtm4/src/main/java/net/ripe/db/nrtm4/persist/NrtmVersionInformationDao.java

```diff
@@ -0,0 +1,92 @@
+package net.ripe.db.nrtm4.persist;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.util.Optional;
+import java.util.UUID;
+
+
+@Repository
+public class NrtmVersionInformationDao {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NrtmVersionInformationDao.class);
+    private final JdbcTemplate jdbcTemplate;
+    private final RowMapper<VersionInformation> rowMapper = (rs, rowNum) ->
```

Comment:
Do you think it's better to create a VersionInfoRowMapper class ? It's more clearer to me rather than creating a private field for it. It might also be needed in few other classes.

Reply:
the idea of keeping it private is to hint that all version row mapping should be in this class. and the idea of putting it all in this class is to be sure that it doesn't get used or mutated in ways that aren't written here -- coz to do so would cause big problems.